src/libYARP_dev/src/devices/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -1847,7 +1847,7 @@ bool ControlBoardWrapper::setRefSpeeds(const double *spds)
                 joints[j_dev] = p->base + j_dev;
             }
 
-            p->pos2->setRefSpeeds(wrapped_joints, joints, &spds[j_wrap]);
+            ret = ret && p->pos2->setRefSpeeds(wrapped_joints, joints, &spds[j_wrap]);
```

Comment:
I would make this: 
~~~
bool ok = p->pos2->setRefSpeeds(wrapped_joints, joints, &spds[j_wrap]);
ret = ret && ok; 
~~~
to preserve the old behaviour in which `setRefSpeeds` is called regardless of the value of `ret`.

Reply:
Right, fixed

Reply:
My personal preference is to keep the change as @damn1 did at the beginning:
`ret = ret && p->pos2->setRefSpeeds...`
This is because whenever the return value is checked, it is done that way.
In this case it's more important to me to have all methods behaving in the same way, rather than keep old behaviour.

Reply:
ok, I restored the original fix https://github.com/robotology/yarp/pull/1387/commits/b2f699428fd6b089b532360e15452a8fc6a88fdb

Reply:
Probably an overkill, but it would be nice to have a line on this in the changelog (in the end it is still a change in the behavior).